app/Http/Controllers/UsersController.php

```diff
@@ -218,21 +188,63 @@ public function show($id, $mode = null)
             ]
         );
 
-        $jsonChunks = [
-            'achievements' => $achievements,
-            'beatmapsets' => $beatmapsets,
-            'currentMode' => $currentMode,
-            'kudosu' => $kudosu,
-            'rankHistory' => $rankHistory,
-            'scores' => $scores,
-            'statistics' => $statistics,
-            'user' => $userArray,
-        ];
+        $statistics = json_item(
+            $user->statistics($currentMode),
+            'UserStatistics',
+            ['rank', 'scoreRanks']
+        );
 
-        return view('users.show', compact(
-            'user',
-            'jsonChunks'
-        ));
+        $rankHistoryData = $user->rankHistories()
+            ->where('mode', Beatmap::modeInt($currentMode))
+            ->first();
+
+        $rankHistory = $rankHistoryData ? json_item($rankHistoryData, 'RankHistory') : [];
+
+        if (Request::is('api/*')) {
+            $userArray['statistics'] = $statistics;
```

Comment:
what's with this subtle difference between api and non-api ಠ_ಠ

Reply:
Because the web side will change to that once the rest of the data is moved to async loading

Reply:
change to... what?

Reply:
the lesser data in the api response.